src/org/zaproxy/zap/view/widgets/WritableFileChooser.java

```diff
@@ -26,11 +28,26 @@ public WritableFileChooser(File currentDirectory) {
 	public void approveSelection() {
 		File selectedFile = getSelectedFile();
 
+		File checkFile = selectedFile;
+		boolean fileExists = checkFile.exists();
+		if (!fileExists) {
+			checkFile = checkFile.getParentFile();
+		}
+		if (checkFile.getUsableSpace() < MINIMUM_SPACE_REQUIREMENT_MB) {
+			int result = JOptionPane.showConfirmDialog(this,
+					Constant.messages.getString("report.write.diskspace.warning.dialog.message"),
+					Constant.messages.getString("report.write.diskspace.warning.dialog.title"),
+					JOptionPane.YES_NO_OPTION);
+			if (result == JOptionPane.NO_OPTION) {
```

Comment:
Maybe this should check that's not `YES_OPTION` instead? That would handle the option `CLOSED_OPTION`, which should behave as `NO_OPTION`.
Thoughts?